nixos/modules/services/network-filesystems/seaweedfs.nix

```diff
@@ -0,0 +1,597 @@
+          User = "seaweedfs";
+          Group = "seaweedfs";
+          StateDirectory = "seaweedfs";
+          RuntimeDirectory = "seaweedfs";
+          Restart = "always";
+          RestartSec = "30s";
+          WorkingDirectory = "${cfg.master.dataDir}";
+          LimitNOFILE = 65535;
+          AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" ];
+        };
+      };
+    })
+
+    (lib.mkIf cfg.volume.enable {
+      systemd.services.seaweedfs-volume = {
+        description = "SeaweedFS Volume Server";
+        after = [
+          "network-online.target"
+          "seaweedfs-master.service"
+        ];
+        wants = [ "network-online.target" ];
+        requires = [ "seaweedfs-master.service" ];
+        wantedBy = [ "multi-user.target" ];
+
+        preStart = ''
+          mkdir -p ${cfg.volume.dataDir}
+          chown -R seaweedfs:seaweedfs ${cfg.volume.dataDir}
+          chmod -R 755 ${cfg.volume.dataDir}
+          ${lib.optionalString (cfg.volume.idxDir != null) ''
+            mkdir -p ${cfg.volume.idxDir}
+            chown -R seaweedfs:seaweedfs ${cfg.volume.idxDir}
+            chmod -R 755 ${cfg.volume.idxDir}
+          ''}
+        '';
+
+        serviceConfig = {
+          ExecStart = ''
+            ${cfg.package}/bin/weed volume \
+              -port=${toString cfg.volume.port} \
+              ${lib.optionalString (cfg.volume.grpcPort != null) "-port.grpc=${toString cfg.volume.grpcPort}"} \
+              -ip=${cfg.volume.ip} \
+              ${lib.optionalString (cfg.volume.ipBind != "") "-ip.bind=${cfg.volume.ipBind}"} \
+              -dir=${cfg.volume.dataDir} \
+              -mserver=${lib.concatStringsSep "," cfg.volume.master} \
+              -max=${toString cfg.volume.maxVolumes} \
+              ${lib.optionalString (cfg.volume.dataCenter != "") "-dataCenter=${cfg.volume.dataCenter}"} \
+              ${lib.optionalString (cfg.volume.rack != "") "-rack=${cfg.volume.rack}"} \
+              ${lib.optionalString (cfg.volume.disk != "") "-disk=${cfg.volume.disk}"} \
+              ${lib.optionalString (cfg.volume.idxDir != null) "-dir.idx=${cfg.volume.idxDir}"} \
+              -index=${cfg.volume.index} \
+              -readMode=${cfg.volume.readMode} \
+              -minFreeSpace=${cfg.volume.minFreeSpace} \
+              -fileSizeLimitMB=${toString cfg.volume.fileSizeLimitMB} \
+              ${
+                lib.optionalString (
+                  cfg.volume.metricsPort != null
+                ) "-metricsPort=${toString cfg.volume.metricsPort}"
+              } \
+              ${
+                lib.optionalString (
+                  cfg.volume.whiteList != [ ]
+                ) "-whiteList=${lib.concatStringsSep "," cfg.volume.whiteList}"
+              }
+          '';
+          User = "seaweedfs";
+          Group = "seaweedfs";
+          StateDirectory = "seaweedfs";
+          RuntimeDirectory = "seaweedfs";
+          Restart = "always";
+          RestartSec = "45s";
```

Comment:
same question as below

Reply:
Have increase time for help cluster 10s is too speed.
So have set 30s master 45s volume 60s filer.